### PR TITLE
501 mftf fails to rerun when storefront customer log out occurs after customer deletion

### PR DESCRIPTION
**Pull request by sagar1dahiwala**
### Description (*)

Updated MFTF Tests that fails to Rerun When Test Logs Out of Storefront After Deleting Customer

### Related Pull Requests
1. https://github.com/magento/partners-magento2b2b/pull/507
2. https://github.com/magento/partners-magento2ee/pull/430

### Fixed Issues (if relevant)

1. Fixes magento/partners-magento2b2b#501

### Manual testing scenarios (*)

1. None

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by danmooney2** [comment]
@magento run Functional Tests CE

File: app/code/Magento/Catalog/Test/Mftf/Test/StorefrontPurchaseProductCustomOptionsDifferentStoreViewsTest.xml [hunk 1]
```diff
@@ -48,6 +48,8 @@
         </before>
 
         <after>
+            <!-- Customer Log Out -->
+            <actionGroup ref="StorefrontCustomerLogoutActionGroup" stepKey="customerLogout"/>
```
Comment (danmooney2) [inline app/code/Magento/Catalog/Test/Mftf/Test/StorefrontPurchaseProductCustomOptionsDifferentStoreViewsTest.xml]: Please use the same stepkey: customerLogoutStorefront

File: app/code/Magento/Swatches/Test/Mftf/Test/StorefrontCustomerCanChangeProductOptionsUsingSwatchesTest.xml [hunk 2]
```diff
@@ -22,8 +22,9 @@
             <createData entity="Simple_US_Customer" after="createCategory" stepKey="createCustomer"/>
         </before>
         <after>
+            <!-- Logout customer -->
+            <actionGroup ref="StorefrontCustomerLogoutActionGroup" stepKey="customerLogoutStorefront"/>
```
Comment (danmooney2) [inline app/code/Magento/Swatches/Test/Mftf/Test/StorefrontCustomerCanChangeProductOptionsUsingSwatchesTest.xml]: Please use the same step key: logoutFromCustomer

**Review by danmooney2 — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Test/Mftf/Test/StorefrontPurchaseProductCustomOptionsDifferentStoreViewsTest.xml (hunk 1 above)
Comment (sagar1dahiwala) [reply app/code/Magento/Catalog/Test/Mftf/Test/StorefrontPurchaseProductCustomOptionsDifferentStoreViewsTest.xml]: updated.

File: app/code/Magento/Swatches/Test/Mftf/Test/StorefrontCustomerCanChangeProductOptionsUsingSwatchesTest.xml (hunk 2 above)
Comment (sagar1dahiwala) [reply app/code/Magento/Swatches/Test/Mftf/Test/StorefrontCustomerCanChangeProductOptionsUsingSwatchesTest.xml]: updated.

**Comment by danmooney2** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Review by danmooney2 — APPROVED** [review]

**Comment by danmooney2** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by danmooney2** [comment]
@magento run Functional Tests CE, Functional Tests EE, Semantic Version Checker

**Comment by dhaecker** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B, Semantic Version Checker

**Review by dhaecker — APPROVED** [review]

**Comment by dhaecker** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B, Semantic Version Checker

**Pull request merged**